src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,256 @@
+
+            return ExecuteWithFailOverPolicyAsync(func, cancellationToken);
+        }
+
+        public async Task<IEnumerable<ConfigurationSetting>> GetConfigurationSettingsAsync(SettingSelector selector, CancellationToken cancellationToken = default)
+        {
+            Page<ConfigurationSetting> settingsPage = null;
+            string continuationToken = null;
+            var result = new List<ConfigurationSetting>();
+
+            async Task<Page<ConfigurationSetting>> func(ConfigurationClient client)
+            {
+                IAsyncEnumerator<Page<ConfigurationSetting>> enumerator = client.GetConfigurationSettingsAsync(selector, cancellationToken)
+                                                                                .AsPages(continuationToken: continuationToken)
+                                                                                .GetAsyncEnumerator();
+
+                Page<ConfigurationSetting> page = enumerator != null && await enumerator.MoveNextAsync() ? enumerator.Current : null;
+                return page;
+            }
+
+            do
+            {
+                settingsPage = await ExecuteWithFailOverPolicyAsync(func, cancellationToken);
+                if (settingsPage != null)
+                {
+                    result.AddRange(settingsPage.Values);
+                    continuationToken = settingsPage.ContinuationToken;
+                }
+            }
+            while (settingsPage != null && settingsPage.ContinuationToken != null);
+
+            return result;
+        }
+
+        public void UpdateSyncToken(Uri endpoint, string syncToken)
+        {
+            this._clients.Single(clientAndState => clientAndState.Endpoint.Host.Equals(endpoint.Host)).Client.UpdateSyncToken(syncToken);
+        }
+
+        private async Task<T> ExecuteWithFailOverPolicyAsync<T>(Func<ConfigurationClient, Task<T>> funcToExecute, CancellationToken cancellationToken = default)
+        {
+            var tasks = new List<Task>();
+            Exception lastException = null;
+            IEnumerable<ConfigurationClientState> clients = GetPrioritizedConfigurationClientList();
+
+            using (var attemptsCancellation = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+            {
+                foreach (ConfigurationClientState client in clients)
+                {
+                    var parallelAwait = Task.Delay(this._parallelRetryInterval, attemptsCancellation.Token);
+                    Task<T> funcTask = funcToExecute(client.Client);
+                    tasks.Add(funcTask);
+                    tasks.Add(parallelAwait);
+
+                    var completedTask = await Task.WhenAny(tasks);
+                    if (!completedTask.Equals(parallelAwait))
+                    {
+                        if (completedTask.Status == TaskStatus.RanToCompletion)
+                        {
+                            client.BackoffEndTime = DateTimeOffset.UtcNow;
```

Comment:
The task that completed may not be current client, right?

Reply:
Good point, we would need to map the task to the client and update the status for the client corresponding to the task that is completed.